plugins/modules/gitlab_milestone.py

```diff
@@ -0,0 +1,498 @@
+        type: str
+        required: true
+      due_date:
+        description:
+          - Milestone due date in YYYY-MM-DD format.
+        type: str
+        required: false
+        default: null
+      start_date:
+        description:
+          - Milestone start date in YYYY-MM-DD format.
+        type: str
+        required: false
+        default: null
+      description:
+        description:
+          - Milestone's description
+        type: str
+        default: null
+'''
+
+
+EXAMPLES = '''
+# same project's task can be executed for group
+- name: Create one milestone
+  community.general.gitlab_milestone:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    project: "group1/project1"
+    milestones:
+      - title: milestone_one
+        start_date: "2024-01-04"
+    state: present
+
+- name: Create many group milestones
+  community.general.gitlab_milestone:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    group: "group1"
+    milestones:
+      - title: milestone_one
+        start_date: "2024-01-04"
+        description: this is a milestone
+        due_date: "2024-02-04"
+      - title: milestone_two
+    state: present
+
+- name: Create many project milestones
+  community.general.gitlab_milestone:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    project: "group1/project1"
+    milestones:
+      - title: milestone_one
+        start_date: "2024-01-04"
+        description: this is a milestone
+        due_date: "2024-02-04"
+      - title: milestone_two
+    state: present
+
+- name: Set or update some milestones
+  community.general.gitlab_milestone:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    project: "group1/project1"
+    milestones:
+      - title: milestone_one
+        start_date: "2024-05-04"
+    state: present
+
+- name: Add milestone in check mode
+  community.general.gitlab_milestone:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    project: "group1/project1"
+    milestones:
+      - title: milestone_one
+        start_date: "2024-05-04"
+    check_mode: true
+
+- name: Delete milestone
+  community.general.gitlab_milestone:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    project: "group1/project1"
+    milestones:
+      - title: milestone_one
+    state: absent
+
+- name: Purge all milestones
+  community.general.gitlab_milestone:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    project: "group1/project1"
+    purge: true
+
+- name: Delete many milestones
+  community.general.gitlab_milestone:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    project: "group1/project1"
+    state: absent
+    milestones:
+      - title: milestone-abc123
+      - title: milestone-two
+'''
+
+RETURN = '''
+milestones:
+  description: Four lists of the milestones which were added, updated, removed or exist.
+  returned: always
```

Comment:
I'm pretty sure this isn't returned when the module fails.
```suggestion
  returned: success
```
Same below.